### ISSUE TITLE
refactoring the `correlation` package

### ISSUE THREAD

**Issue by mattansb** [easystats/correlation#298]
As discussed in out meeting yesterday, the correlation package should be broken down (and built up!) into the following "bits":

### A simple 1:1 correlation function (currently done in `cor_test()`

- input: 2 vectors
- methods: any of the 10~ currently available
    - Pearson correlations can also be Bayesian (via `BayesFactor`)
- output: a tidy data frame - with CIs and p-values

The methods for [tetrachoric, polychoric, and biserial](https://easystats.github.io/correlation/index.html?q=tet#tetrachoric-polychoric-biserial-biweight) correlations can be improved, I think.

### A correlation "matrix" function

- input: a data frame (or data frames) with or without those handy `select` arguments.
    - should also support grouped data frames
- methods: same as for the 1:1 variant 
- output: a tidy (long) data frame 
    - ... that can be transformed into a matrix-like output (currently via the `summary()` method).

### A function for part/partial correlation 
This function will also produce multilevel correlations?

- input: a data frame with those handy `select` arguments to control 
    - Need to be able to control what x/y are and what z are, *and* if z is partialled out from x, y, or both.
    - should also support grouped data frames?
- methods: only Pearson for now?
- output: a tidy (long) data frame 
    - ... that can be transformed into a matrix-like output (currently via the `summary()` method).

### Things to keep

- The current plotting options in `see` are good.
- All the `cor_*()` functions also, I think?
- `is.cor` and `isSquare` (maybe rename to snake-case?)
- `z_fisher()` (rename to `fishers_z`, which is more inline with "named" statistic convention in `effectsize`?)

----

Also welcoming @TomGeva that will be working on this with @bwiernik and myself

**Comment by strengejacke** [comment]
Hi Tom, welcome on board! Great to see you participating in the easystats project! 🎉

**Comment by rempsyc** [comment]
Welcome Tom! :) good luck with all this work 😉

**Comment by DominiqueMakowski** [comment]
Hi @TomGeva good to have you ☺️  Looking forward to that roadmap and don't hesitate to nag us if there's anything

**Comment by mattansb** [comment]
BTW, I have some code and a precompiled model (written for a client) to estimate full Bayesian correlation matrices (and partial correlation matrices) using `{cmdstanr}` (and `{posterior}`). We might want to use that instead of `{BayesFactor}`.

**Comment by TomGeva** [comment]
Happy to join!, thanks for the kind words 😁

**Comment by DominiqueMakowski** [comment]
> We might want to use that instead of {BayesFactor}

Yeah it'd be great to have more flexibility in terms of prior setting and all, assuming it preserves some of the advantages of BayesFactor (you don't need to compile stan stuff, it's fast and it gives BFs out of the box)
